### ISSUE TITLE
Accessing fields of type BaseDatastoreInfo

### ISSUE THREAD

**Issue by BrandonW94** [vmware/govmomi#1255]
The below code is an extract of code I wrote to access all datastore information from VMware using govmomi. I am trying to get the naa of the disk which I believe is found in a VmfsDatastoreInfo struct.  As per ds.Info below it is of the type BaseDatastoreInfo.

I went digging and found how to print the field names of a struct which I have done below. I found that f.Interface() contains the data I need when i=3 and t.Field(3).Name is Vmfs however I dont understand how I can access this easily without needing to use reflect.

```

    c, err := govmomi.NewClient(ctx, u, true)
    if err != nil {
        fmt.Println(err)
    }
    m := view.NewManager(c.Client)

    d, _ := m.CreateContainerView(ctx, c.ServiceContent.RootFolder, []string{"Datastore"}, true) 

    if err != nil {
        log.Fatal(err)
    }

    defer d.Destroy(ctx)

    var dss []mo.Datastore

    err = d.Retrieve(ctx, []string{"Datastore"}, []string{"info", "host"}, &dss)
    if err != nil {
        log.Fatal(err)
    }

    for _, ds := range dss {
        s := reflect.ValueOf(ds.Info).Elem()
        typeOfT := s.Type()
        for i := 0; i < s.NumField(); i++ {
            f := s.Field(i)
            fmt.Println(i, typeOfT.Field(i).Name, f.Type(), f.Interface())
        }
    }
```

the output of the above is:
```
0 DatastoreInfo types.DatastoreInfo {{} DS18008 ds:///vmfs/volumes/56709e3b-e223344a6-dc88-5e5a57b112233/ 3029797437440 69201586814
976 68169720922112 70368744177664 2018-10-11 03:47:33.298499 +0000 UTC }
1 MaxPhysicalRDMFileSize int64 70368744177664
2 MaxVirtualRDMFileSize int64 68169720922112
3 Vmfs *types.HostVmfsVolume &{{{} VMFS DS18008 4397778075648} 1 0 0  63963136 5 5.60 56709e3b-e223344a6-dc88-5e5a57b112233 [{{} naa.xxxxxxxxxxxxxxxxxxxxxxx 1}] false <nil> 0xc01122af10 0xc01122af2d }
```
Long story short I want to know how to get naa.xxxxxxxxxxxxxxxxxxxxxxx without using reflect
